presto-product-tests/src/main/java/com/facebook/presto/tests/IntersectTests.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.tests;
+
+import com.facebook.presto.tests.ImmutableTpchTablesRequirements.ImmutableNationTable;
+import com.teradata.tempto.ProductTest;
+import com.teradata.tempto.Requires;
+import org.testng.annotations.Test;
+
+import static com.facebook.presto.tests.TestGroups.INTERSECT;
+import static com.facebook.presto.tests.TestGroups.QUERY_ENGINE;
+import static com.teradata.tempto.assertions.QueryAssert.Row.row;
+import static com.teradata.tempto.assertions.QueryAssert.assertThat;
+import static com.teradata.tempto.query.QueryExecutor.query;
+import static com.teradata.tempto.query.QueryType.SELECT;
+
+@Requires(ImmutableNationTable.class)
+public class IntersectTests
+        extends ProductTest
+{
+    @Test(groups = {INTERSECT, QUERY_ENGINE})
+    public void intersectWithNonDistinctResults()
+    {
+        assertThat(query("select n_regionkey from nation where n_nationkey < 7 INTERSECT select n_regionkey from nation where n_nationkey > 21"))
+                .containsOnly(row(1), row(3));
+    }
+
+    @Test(groups = {INTERSECT, QUERY_ENGINE})
+    public void intersectWithExplicitDistinctClause()
+    {
+        assertThat(query("select n_regionkey from nation where n_nationkey < 7 INTERSECT DISTINCT select n_regionkey from nation where n_nationkey > 21"))
```

Comment:
`DISTINCT` does not affect results for this query. Please use a query where `DISTINCT` changes the output.